### PR TITLE
Switched to pyproject.toml workflow, removed an unneccessary module import

### PR DESCRIPTION
**Pull request by rogersbw**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+requires-python = ">=3.11"
+readme = "README.md"
+license = {text = "MIT"}
+
+dependencies = [
+    "pandas>=2.2.2",
+    "numpy>=1.26.1",
+    "tensorflow>=2.16.1,<3",
+    "tensorflow-probability==0.24.0",
+    "tf-keras>=2.16,<3",
```
Comment (elray1) [inline pyproject.toml:18]: do we use Keras in this package?  If not, let's remove this dependency.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "matplotlib>=3.9.0",
```
Comment (elray1) [inline pyproject.toml:19]: Do we use matplotlib in the package itself, or just in the documentation?  If just in the documentation, let's move this dependency to a separate block, maybe called "doc" or something.  Becky may know if there's a standard name for this.

**Review by elray1 — CHANGES_REQUESTED** [review]
- Added a couple of comments, mostly minor, but a couple that rendered the code un-runnable.
 - After fixing a couple of those issues locally, the demo worked for me.
 - Could we add something to a README file with info about how to set up the development environment?

File: jacques/kernels.py [hunk 3]
```diff
@@ -117,7 +117,7 @@ def gaussian_kernel(x1, x2, B_chol):
     diffs = diff_x_pairs(x1, x2)
     
     # product (x1 - x2)^T * B_chol
-    diff_chol_prod = tf.expand_dims(tf.matmul(diffs, B_chol), -1)
+    diff_chol_prod = tf.expand_dims(tf.matmul(diffs, B_chol, -1))
```
[the remark this answers is not in the document]
Comment (elray1) [reply jacques/kernels.py:120]: The code did not run for me with this change in place.

**Comment by benwrogers87** [comment]
I believe that I fixed everything listed here. Let me know if it works locally for you!

**Review by elray1 — COMMENTED** [review]
one minor change to readme, otherwise lgtm

**Review by elray1 — DISMISSED** [review]

**Comment by rogersbw** [comment]
I believe we had a reviewed and approved a PR that never got merged. Looking to get the main repo caught up to start pushing changes again.

**Review by elray1 — DISMISSED** [review]
had a couple of questions, but approving in case you think it's good to go

**Review by elray1 — APPROVED** [review]

**Pull request merged**